core/src/main/scala/org/bitcoins/core/wallet/utxo/UTXOSpendingInfo.scala

```diff
@@ -404,32 +551,62 @@ case class P2PKWithTimeoutSpendingInfo(
     }
 }
 
-case class MultiSignatureSpendingInfo(
+sealed trait MultiSignatureSpendingInfo
+    extends RawScriptUTXOSpendingInfoSingle {
+  override def conditionalPath: ConditionalPath =
+    ConditionalPath.NoConditionsLeft
+  override def scriptPubKey: MultiSignatureScriptPubKey
+}
+
+case class MultiSignatureSpendingInfoSingle(
     outPoint: TransactionOutPoint,
     amount: CurrencyUnit,
     scriptPubKey: MultiSignatureScriptPubKey,
-    signers: Vector[Sign],
+    signer: Sign,
     hashType: HashType
-) extends RawScriptUTXOSpendingInfo {
-  require(signers.length >= scriptPubKey.requiredSigs, "Not enough signers!")
+) extends MultiSignatureSpendingInfo
 
-  override def conditionalPath: ConditionalPath =
-    ConditionalPath.NoConditionsLeft
-}
-
-/** Info required for signing a [[ConditionalScriptPubKey]] */
-case class ConditionalSpendingInfo(
+case class MultiSignatureSpendingInfoFull(
     outPoint: TransactionOutPoint,
     amount: CurrencyUnit,
-    scriptPubKey: ConditionalScriptPubKey,
+    scriptPubKey: MultiSignatureScriptPubKey,
     signers: Vector[Sign],
-    hashType: HashType,
-    conditionalPath: ConditionalPath)
-    extends RawScriptUTXOSpendingInfo {
+    hashType: HashType
+) extends RawScriptUTXOSpendingInfo
+    with MultiSignatureSpendingInfo {
+  require(signers.length >= scriptPubKey.requiredSigs,
+          s"Not enough signers!: $this")
+
+  override val requiredSigs: Int = scriptPubKey.requiredSigs
+
```

Comment:
`signer` should be overrun to throw an error so people cannot mistakenly use it

Reply:
Also, then all the `UXTOSpendingInfo`s will nested spending infos should have `signer` be `nestedSpendingInfo.signer`